src/main/java/com/expedia/pitchfork/systems/datadog/forwarder/DatadogSpansDispatcher.java

```diff
@@ -88,16 +95,23 @@ private void flush(List<Span> spans) {
 
         Optional<String> body = serialize(datadogSpans);
 
-        body.ifPresent(it -> {
-            datadogClient.put()
-                    .uri("/v0.3/traces")
-                    .body(fromValue(it))
-                    .retrieve()
-                    .toBodilessEntity()
-                    .subscribe(); // FIXME: reactive
-        });
-
-        spans.clear();
+        try {
+            body.ifPresent(it -> {
+                datadogClient.put()
+                        .uri("/v0.3/traces")
+                        .body(fromValue(it))
+                        .retrieve()
+                        .toBodilessEntity()
+                        .subscribe(); // FIXME: reactive
+
+                successCounter.increment(spans.size());
+            });
+        } catch (Exception e) {
+            failureCounter.increment(spans.size());
+            throw new RuntimeException(e);
```

Comment:
Couldn't you just rethrow the caught exception? Maybe catching `RuntimeException` and just re-throwing it?

Reply:
Yeap I can do that.